challenges/sre/challenge-security.md

```diff
@@ -0,0 +1,202 @@
+whiteboard/quiz interviews so common in the industry. It’s not without the
+downsides - it could take longer than traditional interviews. That said, it's
+our view that this type of challenge gives us a more accurate assessment of your
+ability to work well on the types of projects we’re working on day-to-day here
+at Teleport. [Some of the best teams use coding
+challenges](https://sockpuppet.org/blog/2015/03/06/the-hiring-post/). We
+appreciate your time and are looking forward to hacking on this project
+together.
+
+Please come prepared to the walkthrough having reviewed:
+* [Site Reliability Engineering (SRE) Levels](../../levels/sre.pdf)
+* [Challenge documentation](challenge-security.md)
+
+# Challenge
+
+In this challenge, you will build a Go API server interacting with Redis and a Go service to report on its SLOs. This solution will be deployed to Kubernetes with automated builds, containerization, testing, and secure practices.
+
+## Getting Started
+
+First, create a new private GitHub repository, invite the interview panel as contributors, and share the link with the team in Slack.
+
+> [!IMPORTANT]
+> You’ll have up to 2 weeks from the agreed start date to complete the challenge. Please allow 24-48 hours for each PR submission to give the team ample time for review and feedback during business hours.
+
+### PR Submissions
+The repository you created be used for each PR submission during the challenge.  For each submission, please ensure the following:
+
+* Your submissions meets the requirements of the level you are applying for.
+* Split the work into ~3 pull requests: one for the design document, one for the server code, and one covering service levels and deployment.
+* After submitting, the team will review and provide feedback. Open one PR at a time and wait for two approvals before proceeding with submitting another. You can continue working locally while waiting for feedback on the current PR.
+
+## Design Doc
+The first pull request must be a brief design document that describes how you plan to implement the solution. At Teleport, we prefer Markdown for [our designs](https://github.com/gravitational/teleport/blob/master/rfd/0000-rfds.md).
+
+### Requirements
+
+Please be sure to cover the following design topics:
+
+* API structure
+* Service Levels
+  * Service design
+  * Selected objectives and indicators
+* Developer workflow
+  * Ease of building, running and testing the server
+* Build, security, and deployment
+  * Certificates and ciphers
+  * RBAC and network policy
+  * Selected invariants
+
+### Suggestions
+* Complete the design document within the first week.  This is to ensure you have enough time to work on implementation.
+* Avoid writing an overly detailed design document. 500-1500 words is a good target.
+* We encourage you to ask questions in Slack.
+* Avoid sending draft PRs for feedback as it is difficult to evaluate which parts are draft or complete.
+
+## Implementation
```

Comment:
I think we should sprinkle in more security terms and requirements in each of these sections. Zero-trust. 

E.g. The server should be secured using mTLS, communication between the server and Redis mTLS, communication between SLO service and server (and maybe Redis) mTLS. Maybe even mention thinking about the trust model. Should a client connecting to the server use the same CA as the internal services?

Also, I think L4 should implement mTLS. It's really easy.